### PR TITLE
Preparation for upload to Debian.

### PR DESCRIPTION
**Pull request by smoe**
Hello,

I had volunteered to help bringing mesaflash to Debian (cf. https://github.com/LinuxCNC/linuxcnc/issues/1269). The package builds nicely, is lintian-clean, there is not too much to do. What came to mind:
 * consider being a bit more verbose in the package's description. This gets translated into many languages and search engines should find that - I had worked a bit on it but this is a bit beyond my comfort zone, really.
 * please find a few individuals who you please place on the uploaders line in debian/control, so they can upload new versions to the distribution - I happily mentor through the Debian key signing process.

Best,
Steffen

### CODE REVIEW

**Review by jepler — APPROVED** [review]
This looks fine to me!  Thanks for your efforts.

**Comment by jepler** [comment]
I'm an avid user of debian but I'm not sufficiently involved day to day with linuxcnc to put myself forward as a strong candidate for an uploader. But I'm not sure who is more qualified, either.

File: debian/copyright [hunk 1]
```diff
@@ -1,7 +1,29 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: mesaflash
+Upstream-Contact: LinuxCNC Developers <emc-developers@lists.sourceforge.net>
 Source: https://github.com/micges/mesaflash
 
+Files: *
 Copyright: 2013-2014 Michael Geszkiewicz
 License: GPL-2+
```
Comment (jepler) [inline debian/copyright:8]: Does this also need to be corrected to say "GPL-2.0+"?

Comment (smoe) [reply debian/copyright:8]: Yes, please - forgotten --amend on my side.

File: debian/copyright [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
-License: GPL-2+
+License: GPL-2.0+
 
-The complete text of GPL version 2 can be found in the file
-/usr/share/common-licenses/GPL-2 on Debian systems.
+Files: debian/*
+Copyright: #YEAR# #USERNAME# <#EMAIL#>
+License: GPL-2.0+
+
+License: GPL-2.0+
```
Comment (jepler) [inline debian/copyright:14]: Is this duplication right/intended?

**Comment by smoe** [comment]
> I'm an avid user of debian but I'm not sufficiently involved day to day with linuxcnc to put myself forward as a strong candidate for an uploader. But I'm not sure who is more qualified, either.

Go for it, please. It is the initial step for https://wiki.debian.org/DebianMaintainer and anyone surfacing we can just add with the next update.

File: debian/copyright (hunk 2 above)
Comment (smoe) [reply debian/copyright:14]: Yes. 12 is a reference to a license and 14ff define it. This way you can have two File-Blocks reference the same license.

**Comment by jepler** [comment]
@SebKuzminsky do you want to be listed as an uploader?

**Comment by smoe** [comment]
You can both be uploaders, just put a "," between your "name <email>"s.

**Comment by SebKuzminsky** [comment]
@jepler yes please, I'd like to be listed as an uploader for mesaflash and linuxcnc.

**Comment by smoe** [comment]
@jepler you then please tell me what to upload. A "lastrelease+githash" version is no problem.

**Comment by SebKuzminsky** [comment]
We should talk about bumping the debhelper version from 9 to 10 in master.

We LinuxCNC people have a history of ensuring our release branches build on many different Debian releases (and related distros like Ubuntu, Mint, etc).

Part of the reason for this is "laziness" - building a good realtime kernel using out-of-tree RTAI or Preempt-RT patches took a lot of effort, and we didn't have the volunteer time to support new distros as they came out.  Both sets of realtime patches would also lag mainline kernel versions, sometimes significantly. 

Part of the reason is because we perceived our users as being "upgrade adverse" and often not computer-savvy (and especially not Linux-savvy).  Once they'd built a working CNC machine, we though, they'd not want to jeopardize its stability or take time away from their machining projects to upgrade their distro to a newer version.  So we'd support LinuxCNC releases on the very old distros that we'd successfully built realtime kernels for years ago, so at least they could get LinuxCNC bugfixes (and often whole new major releases) on their old platform.

So specifically, the most recent mesaflash release (3.4.0~pre1) builds on (and we've published packages for):
- Ubuntu 12.04 (Precise)
- Debian 7 (Wheezy) through 10 (Buster)

debhelper 9 works on all those platforms, dh 10 limits us to Stretch and newer, and specifically excludes all our RTAI users.

I'm not saying we shouldn't make this change, I just want to provide background information for @smoe to help explain our particular niche and how it has led to some unorthodox packaging behaviors.  :-)

**Comment by smoe** [comment]
That is all fine. For the upload to Debian unstable we just need to go for the latest and shiniest. But that could be something for the Debian-diff to deal with. Or we could have configure gain some wisdom. Once the package is in testing, you may want to think about a backport to stable (bullseye). Old-stable should also still be in reach.

Debian will not like it that you have its debian folder as part of the upstream source tree, this is for reasons like this, i.e. you want your upstream work unencumbered from packaging peculiarities, thus grant both the Debian and the upstream side more freedom. Let's just ignore that for now as we can work aroun that but mentally also please think about what would change if you had your packaging working in scripts/debian, instead.

**Comment by smoe** [comment]
Quick update: mesaflash was uploaded to Debian, now awaiting a peer review by the FTPmasters of the distribution that every new package experiences. It is seen amongst many others in the list of packages currently being queued at [https://ftp-master.debian.org/new.html](https://ftp-master.debian.org/new.html). With the freeze from the release lifted not too long ago, there is quite a bit of package influx, still, so it will take another while but we will get there.

**Pull request merged**